test/unit/toggle_complete.ts

```diff
@@ -0,0 +1,137 @@
+import { expect } from "chai";
+import * as todoer from "../../src/core/todoer";
+
+describe("Toggling Task Completion", () => {
+  describe("Should toggle an incomplete task to complete", () => {
+    // prettier-ignore
+    const theories = [
+      [ "- [ ] Go shopping",       "- [x] ~~Go shopping~~"     ],
+      [ "- [ ] ~Go shopping~",     "- [x] ~~~Go shopping~~~"   ],
+      [ "- [ ] Go ~~shopping~~",   "- [x] ~~Go ~~shopping~~~~" ],
+      [ "- [ ] ~~Go shopping~~",   "- [x] ~~~~Go shopping~~~~" ],
+      [ "- [ ] [ ] Go shopping",   "- [x] ~~[ ] Go shopping~~" ],
+      [ "- [ ] \t Go shopping \t", "- [x] ~~Go shopping~~"     ],
+      [ "- [ ] ",                  "- [x] ~~~~"                ],
+    ];
+
+    for (const [input, expected] of theories) {
+      it(`Should toggle ${JSON.stringify(input)}`, () => {
+        const actual = todoer.toggleComplete(input);
+        expect(actual).to.be.equal(expected);
+      });
+    }
+  });
+
+  describe("Should toggle a completed task to incomplete", () => {
+    // prettier-ignore
+    const theories = [
+      [ "- [x] ~~Go shopping~~",       "- [ ] Go shopping"     ],
+      [ "- [x] ~~Go ~~shopping~~~~",   "- [ ] Go ~~shopping~~" ],
+      [ "- [x] ~~~Go shopping~~",      "- [ ] ~Go shopping"    ],
+      [ "- [x] ~~~~Go shopping~~~~",   "- [ ] ~~Go shopping~~" ],
+      [ "- [x] ~~[x] Go shopping~~",   "- [ ] [x] Go shopping" ],
+      [ "- [x] ~~ \tGo shopping\t ~~", "- [ ] Go shopping"     ],
+      [ "- [x]  \t~~Go shopping~~\t ", "- [ ] Go shopping"     ],
+      [ "- [x] ~~~~",                  "- [ ] "                ],
+    ];
+
+    for (const [input, expected] of theories) {
+      it(`Should toggle ${JSON.stringify(input)}`, () => {
+        const actual = todoer.toggleComplete(input);
+        expect(actual).to.be.equal(expected);
+      });
+    }
+  });
+
+  describe("Should do nothing when input is not a todo", () => {
+    // prettier-ignore
+    const notTodos = [
+      "Go shopping",
+      "",
+      "\t",
+      "-\t[ ] Go shopping",
+      "-  [ ] Go shopping",
+      "[ ] Go shopping",
+      "- [] Go shopping",
+      "- [ ]Go shopping",
+      "- [x] ~nope~~",
```

Comment:
All these "nope" cases lol.  I'm considering them "not a todo" because they don't have the correct ~~ ~~ on both sides.  What do you think?  Too strict, or OK?